### ISSUE TITLE
Intel XEON performance issue

### ISSUE THREAD

**Issue by moremoneylessproblem** [madMAx43v3r/chia-plotter#787]
serious performance drop @ P3 table stage
minor performance decrease @ other stages

according to the log, each "right" entry on P3 took much longer than previous version. The CPU utilize only 10% on each "right" entry, seems like it only uses single thread to generate certain entries.

**Comment by stevekm** [comment]
possibly related to #785 ?

**Comment by Mattchew86** [comment]
> possibly related to #785 ?

I think so

**Comment by moremoneylessproblem** [comment]
> > possibly related to #785 ?
> 
> I think so

that was on Windows but mine on Linux. My mistake not state it clear at the beginning. Furthermore, in that post, it states that someone cut the time in half in Ubuntu, not on my system. 

the time consuming chart on my system relatively close to the chart in #785 , p3 abnormally slow.

**Comment by gryan315** [comment]
How many physical cores do you have and what value are you using for -r? p3 seems to do best when -r is set at physical cores, and p2 seems to do best when -r is set to logical cores.

**Comment by moremoneylessproblem** [comment]
> How many physical cores do you have and what value are you using for -r? p3 seems to do best when -r is set at physical cores, and p2 seems to do best when -r is set to logical cores.

I’m using identical parameters as I always did with previous version. IMHO, this is a bug, not parameters related issue. 

I have not tested the performance difference as I always set -r no less than logical cores. Will do ASAP and reply you when I get the result.

**Comment by Mattchew86** [comment]
@moremoneylessproblem any devices in raid 0?

I had two NVME's in raid 0 and I improved my P1 & P3 plotting speed by not using RAID

**Comment by stevekm** [comment]
> > > possibly related to #785 ?
> > 
> > 
> > I think so
> 
> that was on Windows but mine on Linux. My mistake not state it clear at the beginning. Furthermore, in that post, it states that someone cut the time in half in Ubuntu, not on my system.
> 
> the time consuming chart on my system relatively close to the chart in #785 , p3 abnormally slow.

@moremoneylessproblem I am not sure that the issue reported in #785 is limited to Windows, Linux users have reported similar issues

**Comment by moremoneylessproblem** [comment]
> @moremoneylessproblem any devices in raid 0?
> 
> I had two NVME's in raid 0 and I improved my P1 & P3 plotting speed by not using RAID

Yes, I have 2 SAS SSD in R0, but still, I don't think this is a hardware issue, the hardware configuration is not changed when I upgrade the plotter to latest version.

we're talking about 50% more plotting time with different plotter version, I don't think any factor rather than plotter itself would make such difference.

**Comment by moremoneylessproblem** [comment]
> How many physical cores do you have and what value are you using for -r? p3 seems to do best when -r is set at physical cores, and p2 seems to do best when -r is set to logical cores.

I don't know what hardware configuration make you think this way, please see attachment.

![36t](https://user-images.githubusercontent.com/64780104/125225483-faf89680-e301-11eb-9993-c8b1f381410f.png)
![72t](https://user-images.githubusercontent.com/64780104/125225490-fcc25a00-e301-11eb-8e32-3444989057e1.png)

**Comment by moremoneylessproblem** [comment]
> @moremoneylessproblem any devices in raid 0?
> 
> I had two NVME's in raid 0 and I improved my P1 & P3 plotting speed by not using RAID

I have different result on the other machine, nvme R0 significantly reduced plotting time.

**Comment by Mattchew86** [comment]
> > @moremoneylessproblem any devices in raid 0?
> > I had two NVME's in raid 0 and I improved my P1 & P3 plotting speed by not using RAID
> 
> Yes, I have 2 SAS SSD in R0, but still, I don't think this is a hardware issue, the hardware configuration is not changed when I upgrade the plotter to latest version.
> 
> we're talking about 50% more plotting time with different plotter version, I don't think any factor rather than plotter itself would make such difference.

I agree with you, but just something you can try until a newer version comes out that fixes it, or someone finds out what's causing it :)

**Comment by moremoneylessproblem** [comment]
> > > @moremoneylessproblem any devices in raid 0?
> > > I had two NVME's in raid 0 and I improved my P1 & P3 plotting speed by not using RAID
> > 
> > 
> > Yes, I have 2 SAS SSD in R0, but still, I don't think this is a hardware issue, the hardware configuration is not changed when I upgrade the plotter to latest version.
> > we're talking about 50% more plotting time with different plotter version, I don't think any factor rather than plotter itself would make such difference.
> 
> I agree with you, but just something you can try until a newer version comes out that fixes it, or someone finds out what's causing it :)

I’m using previous version, still fast and sold. Thank you.

**Comment by stanchiam** [comment]
confirmed, i have 2 pc plotting and one is really slow especially p3, after i copied the chia_plot that was build in the other pc and run it, the slowness disappeared and is now back to normal

**Comment by gryan315** [comment]
> > How many physical cores do you have and what value are you using for -r? p3 seems to do best when -r is set at physical cores, and p2 seems to do best when -r is set to logical cores.
> 
> I don't know what hardware configuration make you think this way, please see attachment.


Maybe several hundred plots made with different settings and configurations on dual xeon E5-2678v3 with extremely consistent results, including plots made with the latest build.

**Comment by moremoneylessproblem** [comment]
> > > How many physical cores do you have and what value are you using for -r? p3 seems to do best when -r is set at physical cores, and p2 seems to do best when -r is set to logical cores.
> > 
> > 
> > I don't know what hardware configuration make you think this way, please see attachment.
> 
> Maybe several hundred plots made with different settings and configurations on dual xeon E5-2678v3 with extremely consistent results, including plots made with the latest build.

to me, set -r with logic core number always faster. But I’ll split it when I plot more than one plot.

**Comment by gryan315** [comment]
In my case, with 24c/48t, my fastest times when plotting in RAM are `-r 32 -u 512` with nvme as -t. At -r 24, phase 3 is slightly faster, and phase 2 is slightly slower, p1 unaffected. At -r 48, phase 2 is noticeably faster, but I start to see left/right slowdowns on phase 3 and it becomes significantly slower. There's an unmerged branch that allows doubling of phase 2 threads which would probably provide the best performance at -r 24, but I haven't had time to test it.

**Comment by moremoneylessproblem** [comment]
> In my case, with 24c/48t, my fastest times when plotting in RAM are `-r 32 -u 512` with nvme as -t. At -r 24, phase 3 is slightly faster, and phase 2 is slightly slower, p1 unaffected. At -r 48, phase 2 is noticeably faster, but I start to see left/right slowdowns on phase 3 and it becomes significantly slower. There's an unmerged branch that allows doubling of phase 2 threads which would probably provide the best performance at -r 24, but I haven't had time to test it.

Thank you very much for sharing. I haven’t try anything bigger than default with -u since I have no idea what it does. A single plot doesn’t max out my CPUs, so I normally  go -r 48 with 48C, two plots at a time. It doubles outcome with less than 10% extra time.
